server/workflowarchive/archived_workflow_server.go

```diff
@@ -149,3 +149,36 @@ func (w *archivedWorkflowServer) DeleteArchivedWorkflow(ctx context.Context, req
 	}
 	return &workflowarchivepkg.ArchivedWorkflowDeletedResponse{}, nil
 }
+
+func (w *archivedWorkflowServer) ListArchivedWorkflowLabel(ctx context.Context, req *workflowarchivepkg.ListArchivedWorkflowLabelRequest) (*wfv1.LabelKeys, error) {
+	labelkeys, err := w.wfArchive.ListWorkflowsLabelKey()
+	if err != nil {
+		return nil, err
+	}
+	return labelkeys, nil
+}
+
+func (w *archivedWorkflowServer) GetArchivedWorkflowLabel(ctx context.Context, req *workflowarchivepkg.GetArchivedWorkflowLabelRequest) (*wfv1.Labels, error) {
+	options := req.ListOptions
+
+	key := ""
+	for _, selector := range strings.Split(options.FieldSelector, ",") {
```

Comment:
This seems wrong to me. `listOptions` should already have ` labelSelector`.

Reply:
you are right